Orange/widgets/visualize/owscatterplotgraph.py

```diff
@@ -598,493 +569,530 @@ def update_tooltip(self, modifiers):
+        update for every property would essentially reset the graph, which
+        can be time consuming.
+        """
+        data = dict(x=x, y=y)
+        for prop in ('pen', 'brush', 'size', 'symbol', 'data'):
+            data[prop] = plot.data[prop]
+        plot.setData(**data)
+
+    def update_coordinates(self):
+        """
+        Trigger the update of coordinates while keeping other features intact.
+
+        The method gets the coordinates by calling `self.get_coordinates`,
+        which in turn calls the widget's `get_coordinate_data`. The number of
+        coordinate pairs returned by the latter must match the current number
+        of points. If this is not the case, the widget should trigger
+        the complete update by calling `reset_graph` instead of this method.
+        """
+        x, y = self.get_coordinates()
+        if x is None or not len(x):
+            return
+        if self.scatterplot_item is None:
+            kwargs = {"x": x, "y": y, "data": np.arange(self.n_points)}
+            self.scatterplot_item = ScatterPlotItem(**kwargs)
+            self.scatterplot_item.sigClicked.connect(self.select_by_click)
+            self.scatterplot_item_sel = ScatterPlotItem(**kwargs)
+            self.plot_widget.addItem(self.scatterplot_item_sel)
+            self.plot_widget.addItem(self.scatterplot_item)
+            self.update_point_props()
         else:
-            size_data = \
-                self.MinShapeSize + \
-                self.scaled_data.get_column_view(self.attr_size)[0][self.valid_data] * \
-                self.point_width
-        nans = np.isnan(size_data)
-        if np.any(nans):
-            size_data[nans] = self.MinShapeSize - 2
-            self.master.Information.missing_size(self.attr_size)
+            self._update_plot_coordinates(self.scatterplot_item, x, y)
+            self._update_plot_coordinates(self.scatterplot_item_sel, x, y)
 
-        # scale sizes because of overlaps
-        if self.attr_size == OWPlotGUI.SizeByOverlap:
-            size_data = np.multiply(size_data, self.overlap_factor)
+        self.update_label_coords(x, y)
+        self.update_density()  # Todo: doesn't work: try MDS with density on
+        self._reset_view(x, y)
+
+    # Sizes
+    def get_sizes(self):
+        """
+        Prepare data for sizes of points in the plot
 
-        return size_data
+        The method is called by `update_sizes`. It gets the sizes
+        from the widget and performs the necessary scaling and sizing.
+
+        Returns:
+            (np.ndarray): sizes
+        """
+        size_column = self.master.get_size_data()
+        if size_column is None:
+            return np.ones(self.n_points) * self.point_width
+        size_column = size_column.copy()
+        size_column -= np.min(size_column)
+        mx = np.max(size_column)
+        if mx > 0:
+            size_column /= mx
+        return self.MinShapeSize + self.point_width * size_column
```

Comment:
Maybe `return self.MinShapeSize + (5 + self.point_width) * size_column`. @ajdapretnar, can you try and see if you like it?

Reply:
Yes, that's better. At least I can see there some difference between points even at the smallest scale.